client/src/charts/canada/canada.js

```diff
@@ -96,62 +101,86 @@ class CanadaGraphBarLegend extends React.Component {
-  }
-  componentDidMount() {
-    this._render();
-  }
-  componentDidUpdate() {
-    this._render();
-  }
-  componentWillUnmount() {
-    this.graph_instance.cleanup();
-  }
-  _render() {
-    const { graph_args, prov_select_callback, data, selected_year_index } =
-      this.props;
-    const { color_scale, years, formatter } = graph_args;
+function NewCanadaGraph({
+  prov_select_callback,
+  data,
+  selected_year_index,
+  graph_args,
+}) {
+  const { color_scale, formatter } = graph_args;
+  const containerRef = useRef();
+  const [containerWidth, setContainerWidth] = useState(null);
 
-    const graph_area_sel = select(
-      ReactDOM.findDOMNode(this.graph_area.current)
-    );
-
-    const ticks = _.map(years, (y) => `${run_template(y)}`);
-
-    this.graph_instance?.cleanup();
+  const windowWidth = useWindowWidth();
 
-    this.graph_instance = new CanadaD3Component(graph_area_sel.node(), {
-      main_color: get_graph_color(1),
-      secondary_color: tertiaryColor,
-      selected_year_index,
-      data,
-      ticks,
-      color_scale,
-      formatter,
-    });
+  useEffect(() => {
+    setContainerWidth(containerRef?.current?.offsetWidth);
+  }, [windowWidth]);
 
+  const [onMouseEnter, onMouseLeave] = useMemo(() => {
+    //TODO: fix hacky closure that holds onto province
     let active_prov = false;
-    this.graph_instance.dispatch.on("dataMouseEnter", (prov) => {
+    function onMouseEnter(datum, provCode) {
       active_prov = true;
-      prov_select_callback(prov);
-    });
-    this.graph_instance.dispatch.on("dataMouseLeave", () => {
-      _.delay(() => {
+      prov_select_callback(provCode);
+    }
+    function onMouseLeave() {
+      setTimeout(() => {
         if (!active_prov) {
           prov_select_callback(null);
         }
       }, 200);
       active_prov = false;
-    });
+    }
+    return [onMouseEnter, onMouseLeave];
+  }, [prov_select_callback]);
 
-    this.graph_instance.render();
-  }
+  const singleYearData = useMemo(() => {
+    const dataForYear = data[selected_year_index];
+    return _.chain(dataForYear)
+      .map((value, provCode) => ({ value, provCode }))
+      .value();
+  }, [selected_year_index, data]);
+
+  const colorScale = useCallback(
+    (datum, _provCode) => {
+      if (!datum?.value) {
+        return "#ccc";
+      }
+      return interpolateBlues(color_scale(datum.value));
+    },
+    [color_scale]
+  );
+
+  const htmlForLabel = useCallback(
+    (datum, provCode) => {
+      const getName =
+        containerWidth > 800
+          ? (provCode) => getProvinceName(provCode, lang)
+          : (provCode) => getProvinceShortName(provCode, lang);
+      if (!datum?.value) {
+        return null;
+      }
+      return `<div>
+      <div>${formatter(datum.value)}</div>
+      <div>${getName(provCode)}</div>
+    </div>`;
+    },
+    [formatter, containerWidth]
+  );
+
+  return (
+    <div ref={containerRef} style={{ height: "700px" }}>
```

Comment:
This height is probably tweakable based on the container width

Reply:
Haha, yeah
![image](https://user-images.githubusercontent.com/11301919/160137872-e810cc8a-c14e-4738-86cf-71a0ed5ba2fc.png)

Is an explicit height actually necessary? I think it'll just scale out till it runs out of width, and height will be whatever that needs. I think that's fine for the panel use-case at least. Otherwise `max-height: "700px"` would also be responsive.